### PR TITLE
Implemented removeMap for iOS Fix for #165

### PR DESCRIPTION
**Pull request by ynunez**
Implemented removeMap for iOS Fix for #165

### CODE REVIEW

File: ios/Plugin/Plugin.m [hunk 1]
```diff
@@ -11,6 +11,7 @@
            CAP_PLUGIN_METHOD(addMarker, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(addMarkers, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(removeMarker, CAPPluginReturnPromise);
+           CAP_PLUGIN_METHOD(removeMap, CAPPluginReturnPromise);
```
Comment (tafelnl) [inline ios/Plugin/Plugin.m:14]: Can you move this line to line 10 instead (right below `updateMap`)?

File: ios/Plugin/Plugin.swift [hunk 2]
```diff
@@ -192,6 +192,21 @@ public class CapacitorGoogleMaps: CustomMapViewEvents {
             }
         }
     }
+    
+    
+    @objc func removeMap(_ call: CAPPluginCall) {
```
Comment (tafelnl) [inline ios/Plugin/Plugin.swift:197]: Can you move this method to right below `updateMap`? This way we keep it consistent with Android Java implementation

File: ios/Plugin/Plugin.swift [hunk 3]
```diff
@@ -192,6 +192,21 @@ public class CapacitorGoogleMaps: CustomMapViewEvents {
...
+        let mapId: String = call.getString("mapId", "")
+
+        DispatchQueue.main.async {
+            guard let customMapView = self.customWebView?.customMapViews[mapId] else {
+                call.reject("map not found")
+                return
+            }
+            
+            (customMapView).view.removeFromSuperview()
+            self.customWebView?.customMapViews.removeValue(forKey: mapId)
```
Comment (tafelnl) [inline ios/Plugin/Plugin.swift:207]: Can you add `call.resolve()` to the end of this method, so that it is resolved correctly.

**Review by tafelnl — CHANGES_REQUESTED** [review]
Looks good in general. Couple of small comments. Additionally, signed commits are required. It's probably easiest if you make sure signing commits is enabled and then squash / redo your commits.

File: ios/Plugin/Plugin.m (hunk 1 above)
Comment (ynunez) [reply ios/Plugin/Plugin.m]: resolved

File: ios/Plugin/Plugin.swift (hunk 2 above)
Comment (ynunez) [reply ios/Plugin/Plugin.swift]: resolved

File: ios/Plugin/Plugin.swift (hunk 3 above)
Comment (ynunez) [reply ios/Plugin/Plugin.swift]: resolved

**Comment by tafelnl** [comment]
Changes look good. Thanks!

Only thing needed still: could you please [sign](https://docs.github.com/en/enterprise-server@3.4/authentication/managing-commit-signature-verification/signing-commits) your commits?

**Comment by jjozwiak** [comment]
Hi @tafelnl , @ynunez 
Is this PR ready to merge into main? removeMap is an essential method I need for a particular use case I'm working on. Thank you both!

**Comment by tafelnl** [comment]
Closed in #185

**Comment by selected-pixel-jameson** [comment]
@tafelnl Thank you!

**Pull request closed without merging**